src/class.cc

```diff
@@ -121,23 +121,24 @@ Thumbnailer::Status Thumbnailer::GenerateAnimation(WebPData* const webp_data) {
   // below the given byte budget.
   int min_quality = minimum_lossy_quality_;
   int max_quality = 100;
+  int middle;
   int final_quality = -1;
   WebPData new_webp_data;
   WebPDataInit(&new_webp_data);
 
   while (min_quality <= max_quality) {
-    int middle = (min_quality + max_quality) / 2;
-
+    middle = (min_quality + max_quality) / 2;
     config_.quality = middle;
-    WebPDataClear(&new_webp_data);
     GenerateAnimationNoBudget(&new_webp_data);
 
     if (new_webp_data.size <= byte_budget_) {
       final_quality = middle;
+      WebPDataClear(webp_data);
```

Comment:
Note: It would be safer to mention in the comment of `GenerateAnimation()` in class.h that `webp_data` is expected to be initialized (otherwise `WebPDataClear()` might free some random memory somewhere because the pointer is undefined).